anesthetic/samples.py

```diff
@@ -271,6 +273,10 @@ def plot_2d(self, axes, *args, **kwargs):
 
                     self.plot(ax_, x, y, plot_type=types[pos], *args,
                               **local_kwargs[pos])
+                elif ax is not None:
+                    ax_ = ax.twin if x == y else ax
+                    ax_.plot([], [])  # need this to increment color cycle
+                    ax_.scatter([], [])  # need this to increment color cycle
 
```

Comment:
I really like this solution -- I've cleaned up the code a little in the latest push, but this is exactly what was needed

Reply:
Haha, yeah, looks neater now.

Reply:
Why do we need to call ``plot`` and ``scatter`` for every loop?

Reply:
They have each their own separate colour cycles, i.e. incrementing the colour for `plot` from `'C0'` to `'C1'` won't affect a call to `scatter` afterwards, `scatter` will use `'C0'`.